src/shader.cpp

```diff
@@ -105,26 +106,70 @@ void Context::glLinkProgram(GLuint program)
 
             auto loc = gl.GetUniformLocation(program, uniform_name.get());
 
-            programr.uniforms.emplace(loc, uniform_info);
+            programInfo.uniforms.emplace(loc, uniform_info);
         }
     }
 }
 
 void Context::glUseProgram(unsigned int handle) {
-    auto it = programs.find(handle);
+    if (handle == 0) {
+        bound_program = 0;
+        return;
+    }
+
+    if (!validate_program_status(handle)) {
+        return;
+    }
+
+    // TODO: add optional performance warning for rebinding the same program
+    //if (handle == bound_program) {
+    //    output_fmt("glUseProgram(program = %u): Program is already bound.", handle);
+    //}
+
+    bound_program = handle;
```

Comment:
maybe in addition to the other comment. this should reflect the value better as well

`current_program_handle` rather than `bound_program`. the latter sounds like an abstraction again, which we don't really need/want, methinks-:thonking:-out-loud-in-my-head

Reply:
Renamed it to `current_program_handle`.